.github/workflows/release.yml

```
@@ -0,0 +1,89 @@
+---
+name: release
+
+on:
+  release:
+    types:
+      - released
+
+# Set a default shell for any run steps. The `-Eueo pipefail` sets
+# errtrace, nounset, errexit, and pipefail. The `-x` will print all
+# commands as they are run. Please see the GitHub Actions
+# documentation for more information:
+# https://docs.github.com/en/actions/using-jobs/setting-default-values-for-jobs
+defaults:
+  run:
+    shell: bash -Eueo pipefail -x {0}
+
+jobs:
+  diagnostics:
+    name: Run diagnostics
+    # This job does not need any permissions
+    permissions: {}
+    runs-on: ubuntu-latest
+    steps:
+      # Note that a duplicate of this step must be added at the top of
+      # each job.
+      - uses: GitHubSecurityLab/actions-permissions/monitor@v1
+        with:
+          # Uses the organization variable unless overridden
+          config: ${{ vars.ACTIONS_PERMISSIONS_CONFIG }}
+      # Note that a duplicate of this step must be added at the top of
+      # each job.
+      - id: harden-runner
+        name: Harden the runner
+        uses: step-security/harden-runner@v2
+        with:
+          egress-policy: audit
+      - id: github-status
+        name: Check GitHub status
+        uses: crazy-max/ghaction-github-status@v4
+      - id: dump-context
+        name: Dump context
+        uses: crazy-max/ghaction-dump-context@v2
+  release:
+    needs:
+      - diagnostics
+    permissions:
+      # We need write permission to move tags.
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: GitHubSecurityLab/actions-permissions/monitor@v1
+        with:
+          # Uses the organization variable unless overridden
+          config: ${{ vars.ACTIONS_PERMISSIONS_CONFIG }}
+      - id: harden-runner
+        name: Harden the runner
+        uses: step-security/harden-runner@v2
+        with:
+          egress-policy: audit
+      - id: extract-semver-parts
+        name: Extract semver parts
+        uses: zyactions/semver@v1
+        with:
+          # This input consists of a newline-separated list of version
+          # prefixes, so in the interest of future expansion we go
+          # ahead and use the YAML multiline literal style indicator.
+          prefixes: |
+            v
+      - id: checkout-code
+        name: Checkout the code
+        uses: actions/checkout@v4
```

Another consistency one.

```suggestion
      - uses: actions/checkout@v4
```

---

I have been trying to add an `id` and `name` for every step since I started working with the GH Actions jazz recently.  What is your reasoning for not doing that?

---

I don't have a reason to _not_ do that, but we should start with cisagov/skeleton-generic and push it down in a consistent manner.

---

My intent was to take care of this inconsistency when I'm able to start using reusable workflows via [cisagov/github-actions-workflows](https://github.com/cisagov/github-actions-workflows).  In the meantime I'd like to continue using `name` and `id` in the new code I am creating.

---

Please see https://github.com/cisagov/skeleton-action-composite/pull/1#discussion_r1974183436.